doc/internal/man3/ossl_punycode_decode.pod

```diff
@@ -0,0 +1,60 @@
+=pod
+
+=head1 NAME
+
+ossl_punycode_decode, ossl_a2ulabel, ossl_a2ucompare
+- internal punycode-related functions
+
+=head1 SYNOPSIS
+
+  #include "crypto/punycode.h"
+
+  int ossl_punycode_decode(const char *pEncoded, const size_t enc_len,
+  unsigned int *pDecoded, unsigned int *pout_length);
+
+  int ossl_a2ulabel(const char *in, char *out, size_t *outlen);
+
+  int ossl_a2ucompare(const char *a, const char *u);
+
+=head1 DESCRIPTION
+
+PUNYCODE encoding introduced in RFCs 3490-3492 is widely used for
+representation of host names in ASCII-only format. Some specifications,
+such as RFC 8398, require comparison of host names encoded in UTF-8 charset.
+
+ossl_a2ulabel() decodes NULL-terminated hostname from PUNYCODE to UTF-8,
+using a provided buffer for output.
+
+ossl_a2ucompare() accepts PUNYCODE encoded NULL-terminated hostnames, decodes
+the 1st argument from PUNYCODE to UTF-8 and compares it with the 2nd one.
```

Comment:
i misunderstood what this was trying to say, and went back to the function to see.  I suggest the following rephrase:

```
ossl_a2ucompare() accepts two NULL-terminated hostnames, decodes the 1st
from PUNYCODE to UTF-8 and compares it with the 2nd one as is.
```

Reply:
Done. Thank you very much for proofreading!